cpp/src/c_api/random.cpp

```diff
@@ -66,10 +66,8 @@ struct select_random_vertices_functor : public cugraph::c_api::abstract_functor
 
       rmm::device_uvector<vertex_t> local_vertices(0, handle_.get_stream());
 
-      if (!multi_gpu || (handle_.get_comms().get_rank() == 0)) {
-        local_vertices = cugraph::select_random_vertices(
-          handle_, graph_view, rng_state_->rng_state_, num_vertices_, false, false);
-      }
+      local_vertices = cugraph::select_random_vertices(
+        handle_, graph_view, rng_state_->rng_state_, num_vertices_, false, false);
 
       if constexpr (multi_gpu) {
```

Comment:
Is this block still necessary?  I think your updates to `cugraph::select_random_vertices` already guarantees that the vertices are shuffled.

Reply:
You're right and deleted the unnecessary shuffle code.